server/src/core/invitation/invitation-service.ts

```diff
@@ -0,0 +1,131 @@
+import { Db, Collection } from 'mongodb';
+import { generateId } from '../util';
+import { Invitation, DbInvitation, InvitationCreateArgs, InvitationService, InvitationStatus } from './types';
+import { createDbOpFailedError, rethrowIfAppError, createResourceNotFoundError, AppError } from '../error';
+import * as validators from './validator';
+import * as messages from '../messages';
+
+const COLLECTION = 'invitations';
+
+export class Invitations implements InvitationService {
+  private db: Db;
+  private collection: Collection<DbInvitation>;
+  private indexesCreated: boolean;
+
+  constructor(db: Db) {
+    this.db = db;
+    this.collection = this.db.collection(COLLECTION);
+    this.indexesCreated = false;
+  }
+
+  async createIndexes(): Promise<void> {
+    if (this.indexesCreated) return;
+
+    try {
+      /*
+      Expire every invitation document 
+      after 172800 seconds (i.e., two days) have passed since creation
+      */
+      await this.collection.createIndex(
+        { expiresAt: 1},
+        { expireAfterSeconds: 172800 }
+      );
```

Comment:
`expiresAt` should contain the actual expiry date, not the date it was created.
Set `expiresAfterSeconds` to 0 (if allowed), otherwise set it 1, that way it's scheduled for deletion immediately when the `expiresAt` date arrives.

Note that, the document can last longer in the db even after `expireAfterSeconds` time has elapsed (see docs: https://docs.mongodb.com/manual/core/index-ttl/#timing-of-the-delete-operation). So we should not rely on it for validity, we only have it to clean up the db. Technically, our app should be able to detect whether an invitation has expired whether or not it's still in the db, that's why the `expiresAt` needs to reflect the actual expiry date.